composer/datasets/in_context_learning_evaluation.py

```diff
@@ -154,6 +154,8 @@ def __init__(
         fewshot_random_seed: int,
         cot_delimiter: str = '',
     ):
+        if not hasattr(tokenizer, 'eos_token_id'):
```

Comment:
this isn't quite the right condition. it should be `if tokenizer.eos_token_id is None`. all tokenizers have the attr (https://github.com/huggingface/transformers/blob/224ab70969d1ac6c549f0beb3a8a71e2222e50f7/src/transformers/tokenization_utils_base.py#L795-L821). Could you add a simple test that if you try to construct the dataset with a tokenizer with a `None` eos token, the error is raised?